### PR TITLE
Safe wrappers

### PR DESCRIPTION
**Pull request by cholcombe973**
I need to run this through a battery of tests to see if this code is legit but I wanted to PR this before it gets any bigger.  I did nearly everything except for the async io calls which aren't easy to do safely.  I implemented a recursive decent parser for ceph's tmap data structure.  I'll do another for omap is necessary.  Other highlights include using bitflags where possible to make it easy to set flags.  There are a few functions where ceph only returns that the buffer we passed to it is too small.  I wasn't sure how many times I should iterate before just failing the operation.  It's something that requires more thought.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,8 +16,11 @@ for development etc. See README.md for details.
 """
 
 [dependencies]
+bitflags = "~0.7"
+byteorder = "~0.5"
 libc = "0.2"
-lsio = "0.1"
```
Comment (cholcombe973) [inline Cargo.toml:20]: I removed lsio because I couldn't find anywhere it was used in the project.

File: src/ceph.rs [hunk 2]
```diff
@@ -104,53 +295,1287 @@ pub fn connect_to_ceph(user_id: &str, config_file: &str) -> Result<rados_t, Rado
...
+                         -> Result<i32, RadosError> {
+    if ctx.is_null() {
+        return Err(RadosError::new("Rados ioctx not created.  Please initialize first".to_string()));
+    }
+    let object_name_str = try!(CString::new(object_name));
+
+    unsafe {
+        let ret_code = rados_read(ctx,
+                                  object_name_str.as_ptr(),
+                                  fill_buffer.as_mut_ptr() as *mut i8,
```
Comment (cholcombe973) [inline src/ceph.rs:803]: Testing has revealed a problem with this.  Ceph returns -34 "unknown error".  Some gdb poking might reveal what is wrong.

**Comment by cholcombe973** [comment]
I haven't throughly tested everything yet.  It seems that anything that's reading from a buffer or just doing an operation seems fine.  Writing back into a supplied buffer seems problematic.

**Comment by cholcombe973** [comment]
@hanscj1 do you have time to review this? I realize it's quite large

**Comment by hanscj1** [comment]
yes. sorry for the delay

**Pull request merged**